src/runtime/pythonexception.cs

```diff
@@ -145,6 +146,42 @@ public string PythonTypeName
             get { return _pythonTypeName; }
         }
 
+        /// <summary>
+        /// Formats this PythonException object into a message as would be printed
+        /// out via the Python console. See traceback.format_exception
+        /// </summary>
+        public string Format()
+        {
+            string res;
+            IntPtr gs = PythonEngine.AcquireLock();
+            if (_pyTB != IntPtr.Zero && _pyType != IntPtr.Zero && _pyValue != IntPtr.Zero)
+            {
+                Runtime.XIncref(_pyType);
+                Runtime.XIncref(_pyValue);
+                Runtime.XIncref(_pyTB);
+                using (PyObject pyType = new PyObject(_pyType))
+                using (PyObject pyValue = new PyObject(_pyValue))
+                using (PyObject pyTB = new PyObject(_pyTB))
+                using (PyObject tb_mod = PythonEngine.ImportModule("traceback"))
+                {
+                    var buffer = new StringBuilder();
+                    var values = tb_mod.InvokeMethod("format_exception", pyType, pyValue, pyTB);
+                    Runtime.PyErr_Clear();
```

Comment:
Normally, this would not need to be here. Since you are using high-level API to call Python code, the API should do all the error handling.